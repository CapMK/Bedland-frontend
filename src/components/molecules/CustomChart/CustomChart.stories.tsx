import type { Meta, StoryObj } from '@storybook/react';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import {
  mockDataForVotingChart,
  mockDataForPaymentChart,
  mockDataForPaymentChartResident,
} from 'utils/mock/chartsMockData';
import CustomChart from './CustomChart';
import { CHART_VARIANTS } from './CustomChart.utils';

const paymentChartTypes = [
  CHART_VARIANTS.voting,
  CHART_VARIANTS.paymentManager,
  CHART_VARIANTS.paymentResident,
];

const meta = {
  title: 'Molecules/CustomChart',
  component: CustomChart,
  tags: ['autodocs'],
  argTypes: {
    data: {},
    size: {},
    variant: {
      name: 'variant',
      type: { name: 'string', required: true },
      description: 'Variant of the charts',
      options: paymentChartTypes,
      control: { type: 'select', default: CHART_VARIANTS.paymentManager },
      table: {
        type: {
          summary: `Variants: ${paymentChartTypes.toString().replaceAll(',', ' | ')}`,
        },
        defaultValue: { summary: CHART_VARIANTS.paymentManager },
      },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <div style={{ width: '30.875rem' }}>{Story()}</div>
      </ThemeProvider>
    ),
  ],
  parameters: { backgrounds: { default: 'light' } },
} satisfies Meta<typeof CustomChart>;

export default meta;
type Story = StoryObj<typeof meta>;

export const PaymentManagerChart: Story = {
  render: (args) => <CustomChart {...args} />,
  args: {
    data: mockDataForPaymentChart,
    size: 14,
    variant: CHART_VARIANTS.paymentManager,
    title: 'February overview for all flats:',
    centerText: 'Total payments',
  },
};

export const PaymentResidentChart: Story = {
  render: (args) => <CustomChart {...args} />,
  args: {
    data: mockDataForPaymentChartResident,
    size: 14,
    variant: CHART_VARIANTS.paymentResident,
    title: 'February overview for all flats:',
    centerText: 'Total payments',
  },
};

export const VotingChart: Story = {
  render: (args) => <CustomChart {...args} />,
  args: {
    data: mockDataForVotingChart,
    size: 14,
    variant: CHART_VARIANTS.voting,
  },
};
