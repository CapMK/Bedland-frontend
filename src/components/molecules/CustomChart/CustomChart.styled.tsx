import styled from 'styled-components';

export const ChartWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
`;
export const ChartContainer = styled.div<{ size: number }>`
  align-items: center;
  display: flex;
  flex-direction: row;
  height: ${({ size }) => size}rem;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
`;

export const DoughnutChartWrapper = styled.div<{ size: number }>`
  height: ${({ size }) => size}rem;
  position: relative;
  width: 100%;
`;

export const DoughnutChartContainer = styled.div`
  height: 100%;
  position: absolute;
  width: 100%;
`;

export const ChartTitle = styled.span`
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  justify-content: start;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
`;
