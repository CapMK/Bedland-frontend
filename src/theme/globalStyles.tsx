import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
html, #root, body{
    box-sizing: border-box;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Source Sans Pro', sans-serif;
}

html, body {
    height: 100%;
}

*, *:before, *:after {
    box-sizing: inherit;
    margin: 0;
    padding: 0;
    border: 0;
    outline: 0;
}

a {
    text-decoration: none;
}

button {
    cursor: pointer;
}

ol, ul, li {
	list-style: none;
}

`;

export default GlobalStyle;
