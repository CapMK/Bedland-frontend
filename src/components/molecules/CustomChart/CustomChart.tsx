import { Doughnut, Bar } from 'react-chartjs-2';
import { Chart, ArcElement, Tooltip, Legend } from 'chart.js/auto';
import { FC, PropsWithChildren } from 'react';
import {
  chartDataAdapter,
  CHARTS_INFORMATIONS,
  CHARTS_OPTIONS,
  CHART_PLUGINS,
  CHART_VARIANTS,
  getSortedValuesOutOfInputData,
} from './CustomChart.utils';
import * as Styled from './CustomChart.styled';
import PaymentChartLegend from '../PaymentChartLegend/PaymentChartLegend';
import { CustomChartProps, IsVotingChart } from './CustomChart.types';

Chart.register(ArcElement, Tooltip, Legend);

const CustomChart: FC<PropsWithChildren<CustomChartProps>> = ({
  data,
  size = 12.5,
  variant,
  title,
  centerText = '',
}) => {
  const { labels, colors } =
    CHARTS_INFORMATIONS.paymentChart[
      variant !== CHART_VARIANTS.voting ? variant : CHART_VARIANTS.paymentManager
    ];
  const doughnutChartRadius = (size * 16) / 2;

  return (
    <Styled.ChartWrapper>
      {variant !== CHART_VARIANTS.voting && !IsVotingChart(data) ? (
        <>
          <Styled.ChartTitle>{title}</Styled.ChartTitle>
          <Styled.ChartContainer size={size}>
            <Styled.DoughnutChartWrapper size={size}>
              <Styled.DoughnutChartContainer>
                <Doughnut
                  redraw
                  updateMode='resize'
                  options={CHARTS_OPTIONS.paymentChart(doughnutChartRadius)}
                  plugins={[CHART_PLUGINS.textOnCenterPlugin(centerText)]}
                  data={chartDataAdapter(labels, colors, getSortedValuesOutOfInputData(data))}
                />
              </Styled.DoughnutChartContainer>
            </Styled.DoughnutChartWrapper>
            <PaymentChartLegend data={data} labels={labels} backgroundColor={colors} />
          </Styled.ChartContainer>
        </>
      ) : (
        <Bar
          redraw
          updateMode='resize'
          options={CHARTS_OPTIONS.votingChart}
          data={chartDataAdapter(
            Object.keys(data),
            CHARTS_INFORMATIONS.votingChart.colors,
            Object.values(data),
          )}
        />
      )}
    </Styled.ChartWrapper>
  );
};

export default CustomChart;
