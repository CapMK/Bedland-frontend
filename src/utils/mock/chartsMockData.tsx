import {
  PaymentChartInputDataType,
  VotingChartInputDataType,
} from '../../components/molecules/CustomChart/CustomChart.types';
import { CHARTS_INFORMATIONS } from '../../components/molecules/CustomChart/CustomChart.utils';

export const mockDataForPaymentChart: PaymentChartInputDataType = {
  'Awaiting payments': 21,
  'Paid on time': 4,
  'Payment overdue': 75,
};

export const mockDataForPaymentChartResident = {
  'Awaiting payments': 25,
  'Paid on time': 75,
};

export const mockDataForVotingChart: VotingChartInputDataType = { YES: 12, NO: 6, ABSTAIN: 20 };

export const mockChartLegendData = {
  labels: CHARTS_INFORMATIONS.paymentChart.paymentManager.labels,
  data: mockDataForPaymentChart,
  backgroundColor: CHARTS_INFORMATIONS.paymentChart.paymentManager.colors,
};
