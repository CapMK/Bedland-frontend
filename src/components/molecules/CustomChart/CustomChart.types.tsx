import { ChartOptions, Plugin } from 'chart.js';
import { HexColor } from 'theme/theme.types';

const ChartsColorsNames = {
  votingChart: ['chartVoting_1stColumn', 'chartVotingNo_2ndColumn', 'chartVoting_3rdColumn'],
  paymentChart: ['chartPaymentPaid', 'chartPaymentOverdue', 'chartPaymentAwaiting'],
} as const;

export type VotingChartColorType = {
  [key in (typeof ChartsColorsNames.votingChart)[number]]: HexColor;
};

export type PaymentChartColorType = {
  [key in (typeof ChartsColorsNames.paymentChart)[number]]: HexColor;
};

export type ChartsColorsTypes = {
  votingChart: VotingChartColorType;
  paymentChart: PaymentChartColorType;
};

export const PaymentChartLabels = ['Payment overdue', 'Awaiting payments', 'Paid on time'] as const;

export type ChartsInformations = {
  paymentChart: {
    paymentManager: {
      labels: (typeof PaymentChartLabels)[number][];
      colors: HexColor[];
    };
    paymentResident: {
      labels: (typeof PaymentChartLabels)[number][];
      colors: HexColor[];
    };
  };
  votingChart: { colors: HexColor[] };
};

export type VotingChartInputDataType = {
  [key: string]: number;
};

export type PaymentChartInputDataType = {
  [key in (typeof PaymentChartLabels)[number]]: number;
};

export type InputDataFormat = VotingChartInputDataType | PaymentChartInputDataType;

export const IsVotingChart = (
  data: VotingChartInputDataType | PaymentChartInputDataType,
): data is VotingChartInputDataType => (data as VotingChartInputDataType).YES !== undefined;

export type ChartsOptions = {
  paymentChart: (radius: number) => ChartOptions<'doughnut'>;
  votingChart: ChartOptions<'bar'>;
};
export type ChartsPlugins = {
  textOnCenterPlugin: (text: string, fontSize?: number) => Plugin<'doughnut'>;
};
export type ChartVariants = ['paymentManager', 'paymentResident', 'voting'];

export type ChartVariantsType = {
  [key in ChartVariants[number]]: `${key}`;
};

export type CustomChartProps = {
  data: InputDataFormat;
  size?: number;
  variant: ChartVariants[number];
  title?: string;
  centerText?: string;
};
