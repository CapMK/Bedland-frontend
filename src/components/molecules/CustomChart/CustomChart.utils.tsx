import { Chart, ChartTypeRegistry } from 'chart.js';
import initialTheme from 'theme/initialTheme';
import { HexColor } from 'theme/theme.types';
import {
  ChartsColorsTypes,
  ChartsInformations,
  ChartsOptions,
  ChartsPlugins,
  ChartVariantsType,
  PaymentChartInputDataType,
  PaymentChartLabels,
} from './CustomChart.types';

export const ChartsColorsValue: ChartsColorsTypes = {
  votingChart: {
    chartVoting_1stColumn: initialTheme.palette.greenLight,
    chartVotingNo_2ndColumn: initialTheme.palette.redLight,
    chartVoting_3rdColumn: initialTheme.palette.blueLight,
  },
  paymentChart: {
    chartPaymentPaid: initialTheme.palette.greenDark,
    chartPaymentOverdue: initialTheme.palette.redLight,
    chartPaymentAwaiting: initialTheme.palette.yellow,
  },
} as const;

export const CHARTS_OPTIONS: ChartsOptions = {
  paymentChart: () => ({
    maintainAspectRatio: false,
    responsive: true,
    plugins: {
      legend: {
        display: false,
      },
      tooltip: { enabled: false },
    },
    elements: {
      arc: {
        borderWidth: 0,
      },
    },
  }),
  votingChart: {
    responsive: true,
    plugins: {
      legend: {
        display: false,
      },
      tooltip: { enabled: false },
    },
    indexAxis: 'x',
    scales: {
      y: {
        beginAtZero: true,
      },
      x: {
        grid: {
          offset: true,
        },
      },
    },
  },
};

export const CHART_PLUGINS: ChartsPlugins = {
  textOnCenterPlugin: (centerText: string, sizeOfTheFont = 0.75) => ({
    id: 'textOnCenter',
    beforeDatasetDraw(chart: Chart<keyof ChartTypeRegistry>) {
      const { ctx } = chart;
      const fontSize = sizeOfTheFont;
      const text = centerText.split(' ');
      const textX = chart.getDatasetMeta(0).data[0].x;
      const textY = chart.getDatasetMeta(0).data[0].y;

      ctx.save();
      ctx.font = `${fontSize}rem sans-serif`;
      ctx.fillStyle = 'black';
      ctx.textBaseline = 'middle';
      ctx.textAlign = 'center';
      ctx.fillText(text[0], textX, textY - sizeOfTheFont * 16);
      ctx.fillText(text[1], textX, textY + 8);
      ctx.restore();
      ctx.save();
    },
  }),
};

export const CHARTS_INFORMATIONS: ChartsInformations = {
  paymentChart: {
    paymentManager: {
      labels: [PaymentChartLabels[0], PaymentChartLabels[1], PaymentChartLabels[2]],
      colors: [
        ChartsColorsValue.paymentChart.chartPaymentOverdue,
        ChartsColorsValue.paymentChart.chartPaymentAwaiting,
        ChartsColorsValue.paymentChart.chartPaymentPaid,
      ],
    },
    paymentResident: {
      labels: [PaymentChartLabels[1], PaymentChartLabels[2]],
      colors: [
        ChartsColorsValue.paymentChart.chartPaymentAwaiting,
        ChartsColorsValue.paymentChart.chartPaymentPaid,
      ],
    },
  },

  votingChart: {
    colors: [
      ChartsColorsValue.votingChart.chartVoting_1stColumn,
      ChartsColorsValue.votingChart.chartVotingNo_2ndColumn,
      ChartsColorsValue.votingChart.chartVoting_3rdColumn,
    ],
  },
};

export const CHART_VARIANTS: ChartVariantsType = {
  paymentManager: 'paymentManager',
  paymentResident: 'paymentResident',
  voting: 'voting',
} as const;

export const chartDataAdapter = (
  labels: string[],
  backgroundColor: HexColor[],
  data: number[],
) => ({
  labels,
  datasets: [
    {
      label: ' ',
      data,
      backgroundColor,
    },
  ],
});

export const getSortedValuesOutOfInputData = (data: PaymentChartInputDataType): number[] => {
  const tempArray = PaymentChartLabels.map((paymentLabels) => data[paymentLabels]).filter(
    (item) => item !== undefined,
  );
  return tempArray;
};
