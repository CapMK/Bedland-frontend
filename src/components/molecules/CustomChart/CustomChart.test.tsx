import { renderWithProviders } from 'utils/test.utils';
import { HexColor } from 'theme/theme.types';
import { mockDataForPaymentChart, mockDataForVotingChart } from 'utils/mock/chartsMockData';
import { vi } from 'vitest';
import CustomChart from './CustomChart';
import { CustomChartProps, PaymentChartInputDataType } from './CustomChart.types';
import { chartDataAdapter, getSortedValuesOutOfInputData } from './CustomChart.utils';

vi.mock('react-chartjs-2', () => ({
  Doughnut: () => null,
  Bar: () => null,
}));

const mockPaymentDataSorted: PaymentChartInputDataType = {
  'Payment overdue': mockDataForPaymentChart['Payment overdue'],
  'Awaiting payments': mockDataForPaymentChart['Awaiting payments'],
  'Paid on time': mockDataForPaymentChart['Paid on time'],
};

const mockCustomChartLabels = ['1Label', '2Label', '3Label'];
const mockCustomChartBackgroundColor: HexColor[] = ['#000000', '#ffffff', '#f06d06'];
const mockCustomChartData: number[] = [24, 36, 40];
const mockReturnCustomChartDataAdapter = {
  labels: mockCustomChartLabels,
  datasets: [
    {
      label: ' ',
      data: mockCustomChartData,
      backgroundColor: mockCustomChartBackgroundColor,
    },
  ],
};

const mockCustomChartPaymentInput: CustomChartProps = {
  data: mockDataForPaymentChart,
  variant: 'paymentManager',
  title: 'Some random title',
  size: 200,
};

const mockCustomChartVotingInput: CustomChartProps = {
  data: mockDataForVotingChart,
  variant: 'paymentManager',
  title: 'Some random title',
  size: 200,
};

describe('Custom Chart', () => {
  const chart = ({ data, variant, title, size }: CustomChartProps) =>
    renderWithProviders(
      <CustomChart data={data} variant={variant} title={title} size={size} />,
      {},
    );

  it('should renders correctly doughnut payment chart for manager', () => {
    expect(chart(mockCustomChartPaymentInput)).toMatchSnapshot();
  });

  it('should renders correctly bar chart for voting', () => {
    expect(chart(mockCustomChartVotingInput)).toMatchSnapshot();
  });

  it('chartDataAdapter function', () => {
    expect(
      chartDataAdapter(mockCustomChartLabels, mockCustomChartBackgroundColor, mockCustomChartData),
    ).toEqual(mockReturnCustomChartDataAdapter);
  });

  it('inputDataForChartConverter payment variant, should return sorted data ', () => {
    expect(getSortedValuesOutOfInputData(mockDataForPaymentChart)).toEqual(
      Object.values(mockPaymentDataSorted),
    );
  });
});
