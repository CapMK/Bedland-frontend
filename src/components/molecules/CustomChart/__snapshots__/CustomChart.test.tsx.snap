// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Custom Chart > should renders correctly bar chart for voting 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<body>
    <div>
      <div
        class="c0"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
    <div
      class="c0"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Custom Chart > should renders correctly doughnut payment chart for manager 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 200rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c3 {
  height: 200rem;
  position: relative;
  width: 100%;
}

.c4 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c5 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c6 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c9 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c8 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c10 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c12 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <span
          class="c1"
        >
          Some random title
        </span>
        <div
          class="c2"
          size="200"
        >
          <div
            class="c3"
            size="200"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            id="legendBox"
          >
            <div
              class="c6"
            >
              <div
                class="c6 c7"
              >
                Payment overdue
              </div>
              <div
                class="c8"
              >
                75%
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c6 c9"
              >
                Awaiting payments
              </div>
              <div
                class="c10"
              >
                21%
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c6 c11"
              >
                Paid on time
              </div>
              <div
                class="c12"
              >
                4%
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 200rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding: 0 0 0rem 2.3rem;
  width: 100%;
}

.c3 {
  height: 200rem;
  position: relative;
  width: 100%;
}

.c4 {
  height: 100%;
  position: absolute;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  font-weight: 600;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  line-height: 140%;
  padding: 1rem 0 1.61rem 1.5rem;
  width: 100%;
}

.c5 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c6 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c9 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c11 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c8 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c10 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c12 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

<div>
    <div
      class="c0"
    >
      <span
        class="c1"
      >
        Some random title
      </span>
      <div
        class="c2"
        size="200"
      >
        <div
          class="c3"
          size="200"
        >
          <div
            class="c4"
          />
        </div>
        <div
          class="c5"
          id="legendBox"
        >
          <div
            class="c6"
          >
            <div
              class="c6 c7"
            >
              Payment overdue
            </div>
            <div
              class="c8"
            >
              75%
            </div>
          </div>
          <div
            class="c6"
          >
            <div
              class="c6 c9"
            >
              Awaiting payments
            </div>
            <div
              class="c10"
            >
              21%
            </div>
          </div>
          <div
            class="c6"
          >
            <div
              class="c6 c11"
            >
              Paid on time
            </div>
            <div
              class="c12"
            >
              4%
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
